django/core/management/__init__.py

```diff
@@ -302,7 +302,17 @@ def execute(self):
                 settings.configure()
 
         if settings.configured:
-            django.setup()
+            # Start the auto-reloading dev server even if the code is broken.
+            if subcommand == 'runserver' and '--noreload' not in self.argv:
```

Comment:
We're in `ManagementUtility.execute()` at this point; we haven't even figured out the list of management commands (because that depends on `INSTALLED_APPS`), let alone instaciated the target `Command` class.